src/app/courses/components/courses-list/courses-list.component.ts

```diff
@@ -8,6 +8,7 @@ import { CoursesService } from '../../services/courses.service';
   selector: 'app-courses-list',
   templateUrl: './courses-list.component.html',
   styleUrls: ['./courses-list.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
```

Comment:
It's not needed here. Moreover, deleting course doesn't work right now because of this.